### PR TITLE
Better document trait objects.

### LINKED ISSUE #171: Move most Trait Object information from types.md to items/traits.md
Also, the way it's written, it looks like every trait impls a trait object on `items/traits.md` right now.

### PR DESCRIPTION
**Pull request by Havvy**
Fixes #171.

### CODE REVIEW

**Comment by Havvy** [comment]
Travis failure is from merging a different PR with a bad example, not from this PR.

File: src/types.md [hunk 1]
```diff
@@ -403,10 +399,31 @@ x = bo(5,7);
...
+
+Two trait object types alias each other if the base traits alias each other and
+if the sets of auto traits are the same. For example,
+`Trait + Send + UnwindSafe` is the same as `Trait + Unwindsafe + Send`.
+
+> Warning: With two trait object types, even when the complete set of traits is
+> the same, if the base traits differ, the type is different. For example,
+> `Send + Sync` is a different type from `Sync + Send`. See [issue 33140].
+
+> Warning: Inlcuding the same auto trait multiple times is allowed, and each
```
Comment (steveklabnik) [inline src/types.md]: small typo: including

File: src/types.md [hunk 2]
```diff
@@ -459,16 +459,18 @@ fn main() {
 In this example, the trait `Printable` occurs as a trait object in both the
 type signature of `print`, and the cast expression in `main`.
 
+### Trait Object Lifetimes
```
Comment (matthewjasper) [inline src/types.md]: "Trait Object Lifetime Bounds"?

Comment (Havvy) [reply src/types.md]: Avoided messing with the verbiage of this section at all (I don't understand what it's conveying yet myself, as I haven't put much thought into it), but that's a better title for the section, so changed it.

**Pull request merged**